src/libfetchers/github.cc

```diff
@@ -8,21 +8,18 @@
 
 namespace nix::fetchers {
 
-std::regex ownerRegex("[a-zA-Z][a-zA-Z0-9_-]*", std::regex::ECMAScript);
-std::regex repoRegex("[a-zA-Z][a-zA-Z0-9_-]*", std::regex::ECMAScript);
-
-struct GitHubInput : Input
+struct GitArchiveInput : Input
```

Comment:
Since this file now contains both GitHub and GitLab functionality, maybe it should be renamed to something like `git-forge(s).cc`? Just a thought to make it somewhat more obvious what the file encompasses, when skimming by filename. (Despite "git forge" being a somewhat recent phrase, I feel it's even less likely someone will know GitLab is also implemented in this file just by reading the filename.)

Reply:
"git forge" seems like a term specific from the emacs community. it certainly sounds more like a brand than a generic term to me. `git platform`?

(we should probably resist bikeshedding the issue beyond this)

Reply:
I don't think it's Emacs-specific, seeing as e.g. https://forgeperf.org/ exists and is in no way related to Emacs (not to mention SourceForge). However, I agree bikeshedding isn't fun. I just thought it would be nice to disambiguate (or inspire the idea of disambiguation).

Reply:
Git hosting is also a nice term, and a more widely-used one AFAIK.